common/src/BasePlugin.php

```diff
@@ -103,19 +103,60 @@ public function isWoocommerceActivated(): bool {
         return class_exists('woocommerce');
     }
 
+    public function getActiveGtinPlugin() {
+        return GtinHandler::getActivePlugin();
+    }
+
     public function getProductMetaKeys(): array {
         global $wpdb;
-        return $wpdb->get_col("
+        $meta_keys = $wpdb->get_col("
             SELECT DISTINCT(pm.meta_key)
             FROM {$wpdb->posts} p
             LEFT JOIN {$wpdb->postmeta} pm ON p.ID = pm.post_id
             WHERE
                 p.post_type = 'product'
                 AND pm.meta_key <> ''
         ");
+        return array_map(
+            function ($value) {
+                return [
+                    'type' => 'meta',
+                    'name' => $value
+                ];
+            },
+            $meta_keys
+        );
     }
 
-    public function hasActiveGtinPlugin(): bool {
-        return GtinHandler::hasActivePlugin();
+    public function getProductKeys() {
+        return array_merge($this->getProductMetaKeys(), $this->getCustomAttributes());
```

Comment:
This is the place where you could add `label` and `option_value` (see my comment in the template).